### PR TITLE
Add test for $val marking values as null

### PR DESCRIPTION
**Pull request by bqrkhn**
Add test cases for $val

### CODE REVIEW

File: src/pg/test/e2e.test.js [hunk 1]
```diff
@@ -439,6 +439,81 @@ describe('pg_e2e', () => {
...
+    //           bar: 'incorrect',
+    //           baz: 'correct'
+    //         }
+    //       }
+    //     },
+    //   ]);
+    // });
+    // afterAll(async () => {
+    //   await store.write(['users', id], null)
+    // })
```
Comment (bqrkhn) [inline src/pg/test/e2e.test.js]: As soon as I try to comment this out and move the code from below to this, it fails with `pg.nothing_written` error. Not sure what I m doing wrong.

File: src/pg/test/e2e.test.js [hunk 2]
```diff
@@ -439,6 +439,81 @@ describe('pg_e2e', () => {
...
+      ]);
+      const res1 = await store.write(['users', id], {
+        settings: {
+          foo: {
+            bar: null,
+          }
+        }
+      });
+      expect(Object.keys(res1.settings.foo)).toEqual(['baz']);
+      await store.write(['users', id], null)
```
Comment (bqrkhn) [inline src/pg/test/e2e.test.js]: If i don't reset this to null, 👇 test cases fails. Instead of `{name: 'A' }` we get `{name: null}`
https://github.com/usegraffy/graffy/blob/7e3f340cf08ffaea9a7a9ab7e23df3e9b9fa358e/src/pg/test/e2e.test.js#L727

File: src/pg/test/e2e.test.js (hunk 1 above)
Comment (bqrkhn) [reply src/pg/test/e2e.test.js]: image

File: src/pg/test/e2e.test.js (hunk 2 above)
Comment (aravindet) [reply src/pg/test/e2e.test.js]: This stuff can be in beforeEach / afterEach

**Comment by aravindet** [comment]
Fixed in #158
